### PR TITLE
Bug/4276 searchpage pluralization fix

### LINKED ISSUE #4276: Search Page - Properly pluralize candidate count
On the search page if the results are a single candidate they are pluralized properly in one place but not in two others.  Don't forget the translate, too!

image

### PR DESCRIPTION
**Pull request by vd1992**
closes #4276 
adjusts wording for pluralization as well as updates fr.json to match

check out -> intl-compile then
assert copy matches expectations for one and other candidate counts (fiddling required to return exactly one candidate from API)
assert the above hold true when adjusting locale to French 

adjusting the French involved me miming the box that was squared in green in the issue screenshot, could be iffy

### CODE REVIEW

**Review by mnigh — COMMENTED** [review]
I have added the different scenarios below that appeared for me locally for comparison sake. First, some questions and suggestions:

1. Should we be using candidat(e) instead of candidat and candidat(e)s instead of candidats (@gobyrne)?

2. I think the French versions for the Zero results strings should be (@gobyrne, can you confirm?):
- Il y a actuellement environ 0 candidats qui répondent à vos critères.
- Résultats : 0 candidats correspondants à vos critères

3. The `candidateCount` is purple in French but black in English. @Jerryescandon [said they should be black](https://talent-cloud.slack.com/archives/CFCE2FSR4/p1665675840658739) and same goes for [Results box `candidateCount` strings](https://talent-cloud.slack.com/archives/CFCE2FSR4/p1665676148303989?thread_ts=1665675840.658739&cid=CFCE2FSR4), so might as well do this now when the IDs are changing (or not, and create another issue).

## Strings

### Zero results

- There are approximately 0 candidates right now who meet your criteria.
- Il y a actuellement environ 0 candidat qui répond à vos critères.
- Results: 0 matching candidates
- Résultats : 0 candidat correspondants à vos critères

### One result

- There is approximately 1 candidate right now who meets your criteria.
- Il y a actuellement environ 1 candidat qui répond à vos critères.
- Results: 1 matching candidate
- Résultats : 1 candidat correspondants à vos critères
- There is 1 matching candidate in this pool
- Il y a 1 candidat qui répondent à vos critères dans ce bassin

### Multiple results

- There are approximately 19 candidates right now who meet your criteria.
- Il y a actuellement environ 19 candidats qui répondent à vos critères.
- Results: 19 matching candidates
- Résultats : 19 candidats correspondants à vos critères
- There are 19 matching candidates in this pool
- Il y a 19 candidats qui répondent à vos critères dans ce bassin

## Screenshots

### Zero results
0-en
0-fr

### One result
1-en
1-fr

### Multiple results
multiple-en
multiple-fr

**Comment by gobyrne** [comment]
@mnigh

1. I don't know... leave as is for now I guess
2. I much prefer the second

**Comment by mnigh** [comment]
> 2\. I much prefer the second

Just to clear, which of these options do you prefer @gobyrne ?

### Option 1

- Il y a actuellement environ 0 candidats qui répondent à vos critères.
- Résultats : 0 candidats correspondants à vos critères

### Option 2

- Il y a actuellement environ 0 candidat qui répond à vos critères.
- Résultats : 0 candidat correspondants à vos critères

**Comment by gobyrne** [comment]
Ah, thanks @mnigh , I guess I misunderstood.

As best I can tell (from a few google searches) both options are grammatically ok and it's up to the writer to decide if we want to emphasise the diversity of the subject by adding an "s" or not.

Let's go with **Option 1**

**Review by mnigh — CHANGES_REQUESTED** [review]
Getting up this French ⛰️ !

1. The deprecated `SearchContainer` component references a French string id, **1xRst5**, that was removed from fr.json. You might be able to use the same English string from the non-deprecated `SearchContainer` component.

2. Can you remove the `primary` tags from the following and then regenerate the French strings:

- https://github.com/GCTC-NTGC/gc-digital-talent/blob/bug/4276-searchpage-pluralization/frontend/talentsearch/src/js/components/search/SearchContainer.tsx#L183-L184
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/bug/4276-searchpage-pluralization/frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx#L203

File: frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx [hunk 1]
```diff
@@ -200,8 +200,8 @@ export const SearchContainer: React.FC<SearchContainerProps> = ({
             {intl.formatMessage(
```
Comment (mnigh) [inline frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx:200]: Is there a reason not to reuse the same object from the non-deprecated SearchContainer component? Then you can get rid of `7Eg+38` in fr.json, I think.

```
            {intl.formatMessage(
              {
                defaultMessage: `{candidateCount, plural,
                  one {Results: <testId>{candidateCount}</testId> matching candidate}
                  other {Results: <testId>{candidateCount}</testId> matching candidates}
                }`,
                id: "Fjikio",
                description:
                  "Heading for total matching candidates in results section of search page.",
              },
              {
                testId,
                candidateCount,
              },
            )}
```

File: frontend/talentsearch/src/js/lang/fr.json [hunk 2]
```diff
@@ -1528,12 +1528,16 @@
     "defaultMessage": "Groupes visés par l’équité en matière d’emploi",
     "description": "Legend for the employment equity checklist"
   },
-  "1xRst5": {
-    "defaultMessage": "Résultats : <primary><testId>{candidateCount}</testId></primary> candidats correspondants à vos critères",
+   "Fjikio": {
+    "defaultMessage": "{candidateCount, plural, =0 {Résultats : <primary><testId>{candidateCount}</testId></primary> candidats correspondants à vos critères} =1 {Résultats : <primary><testId>{candidateCount}</testId></primary> candidat correspondants à vos critères} other {Résultats : <primary><testId>{candidateCount}</testId></primary> candidats correspondants à vos critères}}",
```
Comment (mnigh) [inline frontend/talentsearch/src/js/lang/fr.json:1532]: I think you need to remove the `primary` tags from `7Eg+38` to match the English string.

**Review by mnigh — CHANGES_REQUESTED** [review]
Few more things before I can test.

File: frontend/talentsearch/src/js/lang/fr.json (hunk 2 above)
Comment (vd1992) [reply frontend/talentsearch/src/js/lang/fr.json]: ugh, I'm terrible for the little things 
here https://github.com/GCTC-NTGC/gc-digital-talent/pull/4278/commits/26cbbce231b55b8283f5ba63fdfc31f35a0c1465

File: frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx (hunk 1 above)
Comment (vd1992) [reply frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx:200]: didn't understand what you meant at first
yeah, sure, deprecated isn't intended to be kept around that long anyway
https://github.com/GCTC-NTGC/gc-digital-talent/pull/4278/commits/eabbe0701f9c3ac09c6351b85dfdb738fce324dd

File: frontend/talentsearch/src/js/components/search/SearchContainer.tsx [hunk 3]
```diff
@@ -179,9 +179,11 @@ export const SearchContainer: React.FC<SearchContainerProps> = ({
...
             {intl.formatMessage(
               {
-                defaultMessage:
-                  "Results: <primary><testId>{candidateCount}</testId></primary> matching candidates",
-                id: "1xRst5",
+                defaultMessage: `{candidateCount, plural,
+                  one {Results: <testId>{candidateCount}</testId> matching candidate}
+                  other {Results: <testId>{candidateCount}</testId> matching candidates}
+                }`,
+                id: "Fjikio",
```
Comment (mnigh) [inline frontend/talentsearch/src/js/components/search/SearchContainer.tsx:186]:
```suggestion
                defaultMessage: `{candidateCount, plural,
                  =0 {Results: <testId>{candidateCount}</testId> matching candidates}
                  =1 {Results: <testId>{candidateCount}</testId> matching candidate}
                  other {Results: <testId>{candidateCount}</testId> matching candidates}
                }`,
                id: "aWLo7o",
```

File: frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx [hunk 4]
```diff
@@ -199,9 +199,11 @@ export const SearchContainer: React.FC<SearchContainerProps> = ({
...
               {
-                defaultMessage:
-                  "Results: <primary><testId>{candidateCount}</testId></primary> matching candidates",
-                id: "1xRst5",
+                defaultMessage: `{candidateCount, plural,
+                  one {Results: <testId>{candidateCount}</testId> matching candidate}
+                  other {Results: <testId>{candidateCount}</testId> matching candidates}
+                }`,
+                id: "Fjikio",
```
Comment (mnigh) [inline frontend/talentsearch/src/js/components/search/deprecated/SearchContainer.tsx:206]:
```suggestion
                defaultMessage: `{candidateCount, plural,
                  =0 {Results: <testId>{candidateCount}</testId> matching candidates}
                  =1 {Results: <testId>{candidateCount}</testId> matching candidate}
                  other {Results: <testId>{candidateCount}</testId> matching candidates}
                }`,
                id: "aWLo7o",
```

File: frontend/talentsearch/src/js/lang/fr.json [hunk 5]
```diff
@@ -1528,12 +1528,12 @@
     "defaultMessage": "Groupes visés par l’équité en matière d’emploi",
     "description": "Legend for the employment equity checklist"
   },
-  "1xRst5": {
-    "defaultMessage": "Résultats : <primary><testId>{candidateCount}</testId></primary> candidats correspondants à vos critères",
+   "Fjikio": {
+    "defaultMessage": "{candidateCount, plural, =0 {Résultats : <testId>{candidateCount}</testId> candidats correspondants à vos critères} =1 {Résultats : <testId>{candidateCount}</testId> candidat correspondants à vos critères} other {Résultats : <testId>{candidateCount}</testId> candidats correspondants à vos critères}}",
```
Comment (mnigh) [inline frontend/talentsearch/src/js/lang/fr.json:1532]:
```suggestion
   "aWLo7o": {
    "defaultMessage": "{candidateCount, plural, =0 {Résultats : <testId>{candidateCount}</testId> candidats correspondants à vos critères} =1 {Résultats : <testId>{candidateCount}</testId> candidat correspondant à vos critères} other {Résultats : <testId>{candidateCount}</testId> candidats correspondants à vos critères}}",
```

File: frontend/talentsearch/src/js/lang/fr.json [hunk 6]
```diff
@@ -1569,7 +1569,7 @@
     "description": "Instruction on what to do if user does not have a GC Key."
   },
   "ID8FNk": {
```
Comment (mnigh) [inline frontend/talentsearch/src/js/lang/fr.json:1571]:
```suggestion
  "Bp3HEe": {
```

**Review by mnigh — CHANGES_REQUESTED** [review]
The ⛰️ is steep and rocky! If any of my comments or suggestions need more clarification, feel free to send me a message on Slack and I would be happy to explain.

File: frontend/talentsearch/src/js/lang/fr.json [hunk 7]
```diff
@@ -1569,7 +1569,7 @@
...
-    "defaultMessage": "{candidateCount, plural, one {Il y a actuellement environ <heavyPrimary><testId>{candidateCount}</testId></heavyPrimary> candidat qui répond à vos critères.} other {Il y a actuellement environ <heavyPrimary><testId>{candidateCount}</testId></heavyPrimary> candidats qui répondent à vos critères.} }",
+    "defaultMessage": "{candidateCount, plural, =0 {Il y a actuellement environ <strong><testId>{candidateCount}</testId></strong> candidats qui répondent à vos critères.} =1 {Il y a actuellement environ <strong><testId>{candidateCount}</testId></strong> candidat qui répond à vos critères.} other {Il y a actuellement environ <strong><testId>{candidateCount}</testId></strong> candidats qui répondent à vos critères.} }",
```
[the remark this answers is not in the document]
Comment (vd1992) [reply frontend/talentsearch/src/js/lang/fr.json:1572]: yep yep https://github.com/GCTC-NTGC/gc-digital-talent/pull/4278/commits/c96d07bb4269067dc5ff419a9b87ab1384fa99f9

**Review by mnigh — APPROVED** [review]
The ⛰️ has been summited, nice work @vd1992! 👍

**Pull request merged**